Add Resources feature documentation (#1497)
flowglad/flowglad: platform/docs/features/resources.mdx

---

Resources are claimable capacity entitlements included in subscriptions. They represent countable units that customers can reserve and release—like team seats, API keys, concurrent connections, or projects. Unlike [usage-based billing](/features/usage) which tracks consumption, resources track allocation: who has reserved what.

## The Resource Data Model

Resources let you implement allocation-based pricing models. Here's how the key components work together:

1. **`Resources`**: Define the types of capacity you offer. Each resource has a slug (like `seats` or `api_keys`) and belongs to a pricing model. Think of it as describing *what* can be allocated.

2. **`Subscription Item Features`**: When a customer subscribes to a product, their subscription includes features that grant resource capacity. For example, a "Team Plan" might include 10 seats. This defines *how much* of each resource a customer can claim.

3. **`Resource Claims`**: Individual allocations from a customer's capacity pool. When a user joins a team and takes a seat, that's a claim. Claims can be anonymous (just a count) or named (with an identifier like a user ID).

<Note>

> Resources differ from [Usage](/features/usage) in a key way: usage tracks *consumption* that gets billed, while resources track *allocations* that can be reserved and released. A seat can be claimed and later released when someone leaves the team; a consumed API token is gone forever.

Reviewer: ```suggestion
Resources are different from [Usage](/features/usage): usage tracks *consumption* that gets billed (like tokens), while resources track *allocations* (like seats) that can be reserved and released.
```
